src/channel.rs

```diff
@@ -33,7 +33,7 @@ static RECEIVER: OnceCell<Mutex<Option<StdUnixStream>>> = OnceCell::new();
 pub fn init_channel() -> anyhow::Result<()> {
     let (sender, receiver) = StdUnixStream::pair()?;
 
-    sender.set_nonblocking(true)?;
+    sender.set_nonblocking(false)?;
```

Comment:
Should an option be added to `php.ini` instead of writing it as fixed

Reply:
I agree with the considerations on the main process. As I mentioned, a non-blocking stream fails on `write_all` occasionally (actually, not that occasional). This method (`write_all`) will continuously call `write`, which is not *atomic*. Since we are using a prefix length to serialize and deserialize the data, a `write_all` failure would be unrecoverable (from the stream). IMHO, that is not acceptable either.
A blocking `sender` is not good, but an option does not help. Why do we communicate via a hand-crafted Unix pipe instead of a Rust native channel?

Reply:
Rust native channel means `mpsc`? It can't be used cross process, I think.

Reply:
The problem of `write` is not atomic, I have an idea, disassemble the data into separate packets and assemble them at the receiving end, similar to multiplexing. I will solve this problem in the next PR.

So now keep `nonlocking` true?

Reply:
Then why do we need it to be 'cross process'?

Reply:
> Then why do we need it to be 'cross process'?

Because php-fpm is `multi-process` arch.

Reply:
> If we could not ensure the packet is atomic, recovering from a broken stream would be a challenge.

Yes, but blocking is more undesirable.

Reply:
Would the problem be simpler if we utilize one worker for each fpm first? Anyway, the telemetry data are multiplexed.

Reply:
At first, I used `UnixDatagram`, but I switched to `UnixStream` due to problems with MacOS.

`UnixDatagram` can provide atomic send?

https://github.com/apache/skywalking-php/pull/15

Reply:
IMHO, `UnixDatagram` is a better choice here. A `Datagram` is either sent or not. At least the failure is localized, which simplifies the logic. A blocking sender does not actually solve the problem but makes the behavior predictable. I think we could fix it in another PR. We are actually implementing a local proxy to the backend. I'm not quite sure whether it is a good design.

Reply:
We should not rely on any OS non-blocking methods directly. These operations are somewhat out of control. A buffer could be applied to coordinate the stream, of which the size (or so) is configurable.

Reply:
OK, let's rollback this code, and new a PR to deal with it.

Reply:
Shall we create an issue/task for it?